### ISSUE TITLE
NOT_FOUND in production and "public" folder not being served in development

### ISSUE THREAD

**Issue by dy0gu** [kravetsone/elysia-remix#3]
This issue is just a formality to keep track of [this](https://github.com/elysiajs/elysia/issues/12#issuecomment-2254219918) discussion in a place more suited for it.

In short, the example I give launches a development server which fails to serve the contents of the "public" folder. The same example, when ran with build + production, seems to start serving the "public" folder correctly but now every web page returns NOT_FOUND.

More information about this problem, including details and a minimal reproduction demo, can be found in the [respective repository](https://github.com/dy0gu/elysia-remix-bug-demo).

**Comment by kravetsone** [comment]
> This issue is just a formality to keep track of [this](https://github.com/elysiajs/elysia/issues/12#issuecomment-2254219918) discussion in a place more suited for it.
> 
> In short, the example I give launches a development server which fails to serve the contents of the "public" folder. The same example, when ran with build + production, seems to start serving the "public" folder correctly but now every web page returns NOT_FOUND.
> 
> More information about this problem, including details and a minimal reproduction demo, can be found in the [respective repository](https://github.com/dy0gu/elysia-remix-bug-demo).

![image](https://github.com/user-attachments/assets/84b278f9-3e8a-4dcd-a0ea-b8761f672436)

My mistake is I forgot to update the dependency with an error correction and the semver did not work here

![image](https://github.com/user-attachments/assets/33222c36-17a2-4a8a-9bfd-288d468b0b9d)

**Comment by kravetsone** [comment]
> This issue is just a formality to keep track of [this](https://github.com/elysiajs/elysia/issues/12#issuecomment-2254219918) discussion in a place more suited for it.
> 
> In short, the example I give launches a development server which fails to serve the contents of the "public" folder. The same example, when ran with build + production, seems to start serving the "public" folder correctly but now every web page returns NOT_FOUND.
> 
> More information about this problem, including details and a minimal reproduction demo, can be found in the [respective repository](https://github.com/dy0gu/elysia-remix-bug-demo).

i catch it
its because static plugin and remix handlers server on /* path
But idk how to solve it

**Comment by kravetsone** [comment]
workaround until i publish new version for public folder in dev is override `elysia-connect-middleware` verson:

```json
{
	"name": "test-frontend",
	"private": true,
	"type": "module",
	"scripts": {
		"build": "remix vite:build",
		"dev": "NODE_ENV=development bun src/server.ts",
		"start": "NODE_ENV=production bun src/server.ts"
	},
	"dependencies": {
		"@remix-run/react": "^2.10.3",
		"elysia": "^1.1.2",
		"elysia-remix": "^0.0.1",
		"isbot": "^5.1.13",
		"react": "^18.3.1",
		"react-dom": "^18.3.1"
	},
	"devDependencies": {
		"@remix-run/dev": "^2.10.3",
		"@types/react": "^18.3.3",
		"@types/react-dom": "^18.3.0",
		"vite": "^5.3.4",
		"vite-tsconfig-paths": "^4.3.2"
	},
	"overrides": {
		"elysia-connect-middleware": "0.0.3"
	},
	"engines": {
		"node": ">=20.0.0"
	}
}
```

But I think it makes no sense to release a version without a production build fix

**Comment by dy0gu** [comment]
> > This issue is just a formality to keep track of [this](https://github.com/elysiajs/elysia/issues/12#issuecomment-2254219918) discussion in a place more suited for it.
> > In short, the example I give launches a development server which fails to serve the contents of the "public" folder. The same example, when ran with build + production, seems to start serving the "public" folder correctly but now every web page returns NOT_FOUND.
> > More information about this problem, including details and a minimal reproduction demo, can be found in the [respective repository](https://github.com/dy0gu/elysia-remix-bug-demo).
> 
> i catch it its because static plugin and remix handlers server on /* path But idk how to solve it

Do you think maybe we could look at the [Express Remix Adapter](https://github.com/remix-run/remix/tree/main/packages/remix-express) and see it helps figure this out?

And about your second message, yeah I agree, no problem in holding off until we get a production build fix to resolve that.

**Comment by kravetsone** [comment]
> 
> > > This issue is just a formality to keep track of [this](https://github.com/elysiajs/elysia/issues/12#issuecomment-2254219918) discussion in a place more suited for it.
> > > In short, the example I give launches a development server which fails to serve the contents of the "public" folder. The same example, when ran with build + production, seems to start serving the "public" folder correctly but now every web page returns NOT_FOUND.
> > > More information about this problem, including details and a minimal reproduction demo, can be found in the [respective repository](https://github.com/dy0gu/elysia-remix-bug-demo).
> > 
> > i catch it its because static plugin and remix handlers server on /* path But idk how to solve it
> 
> Do you think we could look at the [Express Remix Adapter](https://github.com/remix-run/remix/tree/main/packages/remix-express) and see it helps figure this out? Maybe seeing how they did it, if they had the same problem?
> 

I guess it doesn't help because routing is different)

But workaround for now is to serve public folder at another path than /

**Comment by wingleeio** [comment]
@kravetsone 

I found a potentially solution to this problem.

```typescript
const clientDirectory = join(buildDirectory, "client");
const glob = new Bun.Glob(`${clientDirectory}/**`);
for (const path of glob.scanSync()) {
	elysia.get(
		path.substring(clientDirectory.length), 
		() => new Response(Bun.file(path))
	);
}
```

Instead of using staticPlugin, you can manually create a route for every file. This has resolved the production issue for me, but you lose the configuration you have with the staticPlugin. In general I have never had success using the staticPlugin for Elysia, so I think this solution could work for now.

**Comment by cortopy** [comment]
> workaround until i publish new version for public folder in dev is override `elysia-connect-middleware` verson:
> 
> ```json
> {
> 	"name": "test-frontend",
> 	"private": true,
> 	"type": "module",
> 	"scripts": {
> 		"build": "remix vite:build",
> 		"dev": "NODE_ENV=development bun src/server.ts",
> 		"start": "NODE_ENV=production bun src/server.ts"
> 	},
> 	"dependencies": {
> 		"@remix-run/react": "^2.10.3",
> 		"elysia": "^1.1.2",
> 		"elysia-remix": "^0.0.1",
> 		"isbot": "^5.1.13",
> 		"react": "^18.3.1",
> 		"react-dom": "^18.3.1"
> 	},
> 	"devDependencies": {
> 		"@remix-run/dev": "^2.10.3",
> 		"@types/react": "^18.3.3",
> 		"@types/react-dom": "^18.3.0",
> 		"vite": "^5.3.4",
> 		"vite-tsconfig-paths": "^4.3.2"
> 	},
> 	"overrides": {
> 		"elysia-connect-middleware": "0.0.3"
> 	},
> 	"engines": {
> 		"node": ">=20.0.0"
> 	}
> }
> ```
> 
> But I think it makes no sense to release a version without a production build fix

I get your point about the new release, but your comment worked for me and it's been more than 3 months. I started by installing the template as per the README, which is broken because of this issue. I nearly gave up as I thought this plugin wasn't working.

Maybe a release is worth it now?

**Comment by kravetsone** [comment]
> > > workaround until i publish new version for public folder in dev is override `elysia-connect-middleware` verson:
> > > 
> > > ```json
> > > {
> > > 	"name": "test-frontend",
> > > 	"private": true,
> > > 	"type": "module",
> > > 	"scripts": {
> > > 		"build": "remix vite:build",
> > > 		"dev": "NODE_ENV=development bun src/server.ts",
> > > 		"start": "NODE_ENV=production bun src/server.ts"
> > > 	},
> > > 	"dependencies": {
> > > 		"@remix-run/react": "^2.10.3",
> > > 		"elysia": "^1.1.2",
> > > 		"elysia-remix": "^0.0.1",
> > > 		"isbot": "^5.1.13",
> > > 		"react": "^18.3.1",
> > > 		"react-dom": "^18.3.1"
> > > 	},
> > > 	"devDependencies": {
> > > 		"@remix-run/dev": "^2.10.3",
> > > 		"@types/react": "^18.3.3",
> > > 		"@types/react-dom": "^18.3.0",
> > > 		"vite": "^5.3.4",
> > > 		"vite-tsconfig-paths": "^4.3.2"
> > > 	},
> > > 	"overrides": {
> > > 		"elysia-connect-middleware": "0.0.3"
> > > 	},
> > > 	"engines": {
> > > 		"node": ">=20.0.0"
> > > 	}
> > > }
> > > ```
> > > 
> > > But I think it makes no sense to release a version without a production build fix
> > 
> > I get your point about the new release, but your comment worked for me and it's been more than 3 months. I started by installing the template as per the README, which is broken because of this issue. I nearly gave up as I thought this plugin wasn't working.
> > 
> > Maybe a release is worth it now?
> 
> Oh sorry i forgot about it... I publish new version with fixes by #7 PR
